apps/api/src/modules/archived-notifications/archived-notifications.service.ts

```diff
@@ -147,4 +151,155 @@ export class ArchivedNotificationsService extends CoreService<ArchivedNotificati
     });
     return archivedNotification;
   }
+
+  async deleteArchivedNotificationsCron(): Promise<void> {
+    try {
+      const enableArchivedNotificationDeletion = this.configService.get<string>(
+        'ENABLE_ARCHIVED_NOTIFICATION_DELETION',
+        'false',
+      );
+
+      if (enableArchivedNotificationDeletion.toLowerCase() === 'true') {
+        this.logger.log('Running archived notification deletion cron task');
+        await this.deleteArchivedEntriesAndGenerateCsvBackup();
+        this.logger.log(`Archive notification deletion cron task completed`);
+      } else {
+        this.logger.log('Archived Notification Deletion Cron is disabled');
+      }
+    } catch (error) {
+      this.logger.error(`Cron job failed: ${error.message}`, error.stack);
+      throw error;
+    }
+  }
+
+  async deleteArchivedEntriesAndGenerateCsvBackup(): Promise<void> {
+    try {
+      const deleteArchivedNotificationsOlderThan = this.configService.get<string>(
+        'DELETE_ARCHIVED_NOTIFICATIONS_OLDER_THAN',
+        '90d',
+      );
+
+      const maxRetentionMs = ms('10y');
+      const retentionDurationMs = ms(deleteArchivedNotificationsOlderThan);
+
+      // Guard rails
+      if (
+        typeof retentionDurationMs !== 'number' ||
+        retentionDurationMs < 0 ||
+        retentionDurationMs > maxRetentionMs
+      ) {
+        throw new Error(
+          'Invalid retention period. It must be a positive duration not exceeding 10 years.',
+        );
+      }
+
+      const cutoffTimestamp = new Date(Date.now() - retentionDurationMs);
+
+      const queryRunner = this.dataSource.createQueryRunner();
+      await queryRunner.connect();
+      await queryRunner.startTransaction();
+
+      try {
+        // Fetch entries to delete
+        const archivedEntries = await queryRunner.manager.find(ArchivedNotification, {
+          where: {
+            createdOn: LessThan(cutoffTimestamp),
+            status: Status.ACTIVE,
+          },
+          order: {
+            createdOn: 'DESC',
+          },
+        });
+
+        if (archivedEntries.length === 0) {
+          this.logger.log('No entries to archive.');
+          await queryRunner.commitTransaction();
+          return;
+        }
+
+        const idsToDelete = archivedEntries.map((entry) => entry.id);
```

Comment:
This will flood the log. It is fine just say the number of records deleted and may be the start and end record since the ids will go mostly in a serial manner.